### PR TITLE
feat: add C `ndarray` API and refactor `blas/ext/base/dnannsumors`

### PR DESCRIPTION
**Pull request by headlessNode**
Resolves none.

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds C `ndarray` API and refactor/cleanup `blas/ext/base/dnannsumors`

## Related Issues

> Does this pull request have any related issues?

No. 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumors/README.md [hunk 1]
```diff
@@ -166,6 +166,132 @@ console.log( out );
...
+### Examples
+
+```c
+#include "stdlib/blas/ext/base/dnannsumors.h"
+#include "stdlib/blase/base/shared.h"
+#include <stdio.h>
+
+int main( void ) {
+    // Create a strided array:
+    const double x[] = { 1.0, 2.0, -3.0, -4.0, 5.0, -6.0, -7.0, 8.0, 0.0/0.0, 0.0/0.0 };
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumors/README.md]:
```suggestion
    const double x[] = { 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 0.0/0.0, 0.0/0.0 };
```

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumors/src/main.c [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	double sum;
+	CBLAS_INT ix;
+	CBLAS_INT i;
+
+	sum = 0.0;
+	ix = offsetX;
+	*n = 0;
+	if ( N <= 0 ) {
+		return sum;
+	}
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumors/src/main.c]:
```suggestion
	sum = 0.0;
	*n = 0;
	if ( N <= 0 ) {
		return sum;
	}
	ix = offsetX;
```

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumors/src/main.c [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	if ( strideX == 0 ) {
+		if ( stdlib_base_is_nan( X[ 0 ] ) ) {
+			return sum;
+		}
+		sum = X[ 0 ] * N;
+		*n += N;
+		return sum;
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumors/src/main.c]:
```suggestion
	if ( strideX == 0 ) {
		if ( stdlib_base_is_nan( X[ ix ] ) ) {
			return sum;
		}
		sum = X[ ix ] * N;
		*n += N;
		return sum;
```

**Comment by headlessNode** [comment]
@kgryte same CI failure after merge. It was not present before.

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumors/src/main.c [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/strided/base/stride2offset.h"
+#include "stdlib/blas/ext/base/dnannsumors.h"
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumors/src/main.c]:
```suggestion
#include "stdlib/blas/ext/base/dnannsumors.h"
#include "stdlib/strided/base/stride2offset.h"
```

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumors/src/main.c [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	double sum;
+	CBLAS_INT ix;
+	CBLAS_INT i;
+
+	sum = 0.0;
+	if ( N <= 0 ) {
+		return sum;
+	}
+	ix = offsetX;
+	*n = 0;
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumors/src/main.c]:
```suggestion
	sum = 0.0;
	*n = 0;
	if ( N <= 0 ) {
		return sum;
	}
	ix = offsetX;
```

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumors/README.md [hunk 6]
```diff
@@ -58,7 +58,7 @@ The function has the following parameters:
 -   **out**: output [`Float64Array`][@stdlib/array/float64] whose first element is the sum and whose second element is the number of non-NaN elements.
 -   **strideOut**: index increment for `out`.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumors/README.md]:
```suggestion
-   **strideOut**: stride length for `out`.
```

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumors/README.md [hunk 7]
```diff
@@ -166,6 +166,132 @@ console.log( out );
...
+
+double v = stdlib_strided_dnannsumors( 4, x, 1, &n );
+// returns 7.0
+```
+
+The function accepts the following arguments:
+
+-   **N**: `[in] CBLAS_INT` number of indexed elements.
+-   **X**: `[in] double*` input array.
+-   **strideX**: `[in] CBLAS_INT` index increment for `X`.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumors/README.md]:
```suggestion
-   **strideX**: `[in] CBLAS_INT` stride length for `X`.
```

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumors/README.md [hunk 8]
```diff
@@ -166,6 +166,132 @@ console.log( out );
...
+
+double v = stdlib_strided_dnannsumors_ndarray( 4, x, 1, 0, &n );
+// returns 7.0
+```
+
+The function accepts the following arguments:
+
+-   **N**: `[in] CBLAS_INT` number of indexed elements.
+-   **X**: `[in] double*` input array.
+-   **strideX**: `[in] CBLAS_INT` index increment for `X`.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumors/README.md]:
```suggestion
-   **strideX**: `[in] CBLAS_INT` stride length for `X`.
```

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumors/lib/ndarray.js [hunk 9]
```diff
@@ -30,10 +30,10 @@ var isnan = require( '@stdlib/math/base/assert/is-nan' );
 *
 * @param {PositiveInteger} N - number of indexed elements
 * @param {Float64Array} x - input array
-* @param {integer} strideX - `x` stride length
+* @param {integer} strideX - `x` index increment
 * @param {NonNegativeInteger} offsetX - `x` starting index
 * @param {Float64Array} out - output array
-* @param {integer} strideOut - `out` stride length
+* @param {integer} strideOut - `out` index increment
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumors/lib/ndarray.js]:
```suggestion
* @param {integer} strideOut - `out` stride length
```

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumors/lib/ndarray.native.js [hunk 10]
```diff
@@ -32,10 +30,10 @@ var addon = require( './dnannsumors.native.js' );
 *
 * @param {PositiveInteger} N - number of indexed elements
 * @param {Float64Array} x - input array
-* @param {integer} strideX - `x` stride length
+* @param {integer} strideX - `x` index increment
 * @param {NonNegativeInteger} offsetX - `x` starting index
 * @param {Float64Array} out - output array
-* @param {integer} strideOut - `out` stride length
+* @param {integer} strideOut - `out` index increment
 * @param {NonNegativeInteger} offsetOut - `out` starting index
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumors/lib/ndarray.native.js]:
```suggestion
* @param {integer} strideX - `x` stride length
* @param {NonNegativeInteger} offsetX - `x` starting index
* @param {Float64Array} out - output array
* @param {integer} strideOut - `out` stride length
* @param {NonNegativeInteger} offsetOut - `out` starting index
```

File: lib/node_modules/@stdlib/blas/ext/base/dnannsumors/src/main.c [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+	CBLAS_INT ox = stdlib_strided_stride2offset( N, strideX );
+	return API_SUFFIX(stdlib_strided_dnannsumors_ndarray)( N, X, strideX, ox, n );
+}
+
+/**
+* Computes the sum of double-precision floating-point strided array elements, ignoring `NaN` values and using ordinary recursive summation and alternative indexing semantics.
+*
+* @param N        number of indexed elements
+* @param X        input array
+* @param strideX  index increment
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsumors/src/main.c]:
```suggestion
* @param strideX  stride length
```

**Review by Planeshifter — APPROVED** [review]
Looks all good to me.

**Pull request merged**